### ISSUE TITLE
Module parse failed: Unexpected token with react

### ISSUE THREAD

**Issue by D4nte** [status-im/js-waku#165]
## Problem

When using js-waku 0.3.0 with react:

```
Failed to compile.

./node_modules/js-waku/build/module/lib/waku_relay/index.js 228:16
Module parse failed: Unexpected token (228:16)
File was processed with these loaders:
 * ./node_modules/react-scripts/node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|       }
| 
>       meshPeers?.forEach(peer => {
|         toSend.add(peer);
|       });
```

It seems that babel has an issue with optional chaining (`?` before `.`).

The issue is not systematic. Indeed, it seems to have been fixed in [babel v7.13.2](https://github.com/babel/babel/releases/tag/v7.13.12).

when creating a new project with `npx create-react-app` on Friday and installing js-waku 0.2.0 the problem did not appear and the following versions of babel were loaded:

```
▶ npm ls @babel/preset-env
waku-demo@0.1.0 /home/froyer/src/waku-demo
└─┬ react-scripts@4.0.3
  ├─┬ @svgr/webpack@5.5.0
  │ └── @babel/preset-env@7.14.2
  ├─┬ babel-preset-react-app@10.0.0
  │ └── @babel/preset-env@7.12.1
  └─┬ workbox-webpack-plugin@5.1.4
    └─┬ workbox-build@5.1.4
      └── @babel/preset-env@7.14.2 deduped
```

However, on Monday 17 May, creating a new project and installing js-waku 0.3.0, the problem appeared:

```
▶ npm ls @babel/preset-env
waku-pres@0.1.0 /home/froyer/src/waku-pres
└─┬ react-scripts@4.0.3
  ├─┬ @svgr/webpack@5.5.0
  │ └── @babel/preset-env@7.12.17
  ├─┬ babel-preset-react-app@10.0.0
  │ └── @babel/preset-env@7.12.1
  └─┬ workbox-webpack-plugin@5.1.4
    └─┬ workbox-build@5.1.4
      └── @babel/preset-env@7.12.17 deduped
```

This time, some older babel versions got loaded.

## Use `--use-npm` flag

This issue does not seem to occur/is less frequent if the `--use-npm` flag is properly used.
Indeed, if you intend to use `npm` for your project (and not `yarn`), be sure to pass `--use-npm` when creating your project:

```sh
npx create-react-app --use-npm <app name>
```

By default, `create-react-app` setup the project for `yarn` usage.

## npm: clean and retry

Sometimes this just works:

```shell
rm -rf node_modules package-lock.json yarn.lock 
npm install
```

### npm: Force babel 7.14

To fix the issue, one can force the installation of babel 7.14 doing the following steps:

```shell
npm i --save-dev @babel/preset-env@7.14
rm -rf node_modules package-lock.json
npm install
```

babel 7.14 should now show (and the compilation error should be gone):

```shell
▶ npm ls @babel/preset-env               
waku-pres@0.1.0 /home/froyer/src/waku-pres
├── @babel/preset-env@7.14.2
└─┬ react-scripts@4.0.3
  ├─┬ @svgr/webpack@5.5.0
  │ └── @babel/preset-env@7.14.2 deduped
  ├─┬ babel-preset-react-app@10.0.0
  │ └── @babel/preset-env@7.12.1
  └─┬ workbox-webpack-plugin@5.1.4
    └─┬ workbox-build@5.1.4
      └── @babel/preset-env@7.14.2 deduped
```

### yarn

yarn does not seem to pull older version of babel so just using yarn is another workaround:

```shell
rm -rf node_modules package-lock.json
yarn
▶ yarn info @babel/preset-env | head -n10
yarn info v1.22.10
{
  name: '@babel/preset-env',
  description: 'A Babel preset for each environment.',
  'dist-tags': {
    latest: '7.14.2'
  },
  versions: [
    '7.0.0-beta.4',
    '7.0.0-beta.5',
```

**Comment by D4nte** [comment]
Issue created on create-react-app repo: https://github.com/facebook/create-react-app/issues/10975

**Comment by D4nte** [comment]
Moved to icebox as little can be done from js-waku side.

**Comment by D4nte** [comment]
This seems resolved with react-scripts 5.0.0
